src/attr.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt;
+use std::borrow::Cow;
+
+use escape::Escaped;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Attribute<'a> {
+    pub name: Cow<'a, str>,
+    pub value: Cow<'a, str>,
```

Comment:
I think it'd be good to have a `new` method for `Attribute` and make its fields private. Then we could check if the name is valid, as well as silently convert underscores to hyphens.